### PR TITLE
feat: add CheckoutEmailAddress component

### LINKED ISSUE #120: Checkout Email Address Component
# Guest or Account Email Display Component
## Overview / Summary
Create a component that will display the email address associated with an order, either the guest email or if signed in, the account email address.

## Expected use cases
This component will be used whenever displaying the customer's email address in the checkout

### Where should this component live
This component should be created in the component library.

## Images
image
image

```jsx
// Pseudo-markup
<div>
  <!-- if isAccountEmail -->
  Signed in as
  <!-- endif -->
  <b>email@example.com</b>
</div>
```

### Props
```js
propTypes: {
  emailAddress: PropTypes.string,
  isAccountEmail: PropTypes.bool
}
```

### Render Criteria
- This component should be rendered on the checkout page above the checkout actions when there is an email address associated with the account or cart

### Normal
As a guest the email address should be rendered.
https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f4fa90d3b184213318da

If the email address is associated with an account, the email address should be prefixed with "Signed in as "
https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f470eefa9c371dd3c488

### Empty
If the email address does not exist, this component should not be rendered.

### PR DESCRIPTION
**Pull request by kieckhafer**
Resolves #120 
Impact: **minor**  
Type: **feature**

## Component
CheckoutEmailAddress component adds a display component for the customers email address. If the user is a registered user, the text "Signed in as" will be displayed along with 

## Screenshots
checkoutemailaddress_ _reaction_design_system

checkoutemailaddress_ _reaction_design_system

## URL
https://deploy-preview-137--stoic-hodgkin-c0179e.netlify.com/#!/CheckoutEmailAddress

## Breaking changes
None

## Testing
Make sure components match designs [1](https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f470eefa9c371dd3c488) [2](https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f4fa90d3b184213318da)

### CODE REVIEW

**Comment by machikoyasuda** [comment]
🚀 Preview deployed

Built with commit 84f058685acba90068bc4c9ba85e274117649554

https://deploy-preview-137--stoic-hodgkin-c0179e.netlify.com

File: package/src/components/CheckoutEmailAddress/v1/CheckoutEmailAddress.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+`;
+
+const StyledSpan = styled.span`
+  font-weight: ${applyTheme("font_weight_bold")};
+`;
+
+class CheckoutEmailAddress extends Component {
+  static propTypes = {
+    emailAddress: PropTypes.string,
+    isAccountEmail: PropTypes.bool
```
Comment (willopez) [inline package/src/components/CheckoutEmailAddress/v1/CheckoutEmailAddress.js]: Lets make these two props required, other than that looks good.

File: package/src/components/CheckoutEmailAddress/v1/CheckoutEmailAddress.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { applyTheme } from "../../../utils";
+
+const StyledDiv = styled.div`
+  color: ${applyTheme("color_coolGrey500")};
+  font-size: ${applyTheme("font_size_small")};
```
Comment (willopez) [inline package/src/components/CheckoutEmailAddress/v1/CheckoutEmailAddress.js]: set font family: `font-family: ${applyTheme("font_family")};`

**Comment by willopez** [comment]
@kieckhafer I adde some missing styles to other components in this PR. This looks go to go, once @rymorgan approves it.

**Comment by willopez** [comment]
@kieckhafer not sure why the snapshot test for QuantityInput is failing in cicle CI, it does not fail locally.

**Pull request merged**